File: 06.adapter/adapter.java
// Interfaz objetivo
interface PaymentProcessor {
    void processPayment(String details);
}

// Clase existente incompatible
class LegacyPaymentSystem {
    void makePayment(String data) {
        System.out.println("Pago procesado: " + data);
    }
}

// Adaptador
class PaymentAdapter implements PaymentProcessor {
    private LegacyPaymentSystem legacyPaymentSystem;

    public PaymentAdapter(LegacyPaymentSystem legacyPaymentSystem) {
        this.legacyPaymentSystem = legacyPaymentSystem;
    }

    @Override
    public void processPayment(String details) {
        legacyPaymentSystem.makePayment(details);
    }
}

// Uso
public class adapter {
    public static void main(String[] args) {
        LegacyPaymentSystem legacySystem = new LegacyPaymentSystem();
        PaymentProcessor adapter = new PaymentAdapter(legacySystem);
        adapter.processPayment("Detalles del pago");
    }
}

File: 06.adapter/adapter.js
// Interfaz objetivo
class PaymentProcessor {
    processPayment(details) {
        throw new Error("Método no implementado");
    }
}

// Clase existente incompatible
class LegacyPaymentSystem {
    makePayment(data) {
        console.log(`Pago procesado: ${data}`);
    }
}

// Adaptador
class PaymentAdapter extends PaymentProcessor {
    constructor(legacySystem) {
        super();
        this.legacySystem = legacySystem;
    }

    processPayment(details) {
        this.legacySystem.makePayment(details);
    }
}

// Uso
const legacySystem = new LegacyPaymentSystem();
const adapter = new PaymentAdapter(legacySyste